files/bin/xdebug-start

```diff
@@ -6,19 +6,35 @@ SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 # Disable is done on exit, this means that no matter what happens (eg the
 # user doing a ^C) we will attempt to re-disable xdebug.
-disable() {
+function disable {
     echo '- disabling xdebug' && \
     phpdismod xdebug
+    if is_docker_for_mac; then 
+        phpdismod xdebug-docker-for-mac 
+    fi
     echo '- reloading php-fpm'
     pkill -HUP 'php-fpm'
     exit $?
 }
+
+# We detect docker-for-mac by resolving a hostname that should only be
+# resolvable on docker-for-mac.
+# See https://docs.docker.com/docker-for-mac/networking
+function is_docker_for_mac {
+    [[ -f /.dockerenv ]] && getent hosts host.docker.internal > /dev/null
```

Comment:
Makes me uneasy. There's an issue asking for the same hostname on Linux: https://github.com/docker/for-linux/issues/264 . Secondly:

```shell
~ ▶ docker run --rm -it alpine sh -c "ls -l /.dockerenv"
-rwxr-xr-x    1 root     root             0 Oct  4 06:10 /.dockerenv
~ ▶ 
```

Reply:
The script can get the ip of the host with `ip route|awk '/default/ { print $3 }'` and write to the PHP "module" file. Then it would work for both platforms.

Reply:
Then make the script use the host.docker.internal if it's available, and fall back to the gateway if it is.

> I could easily see Docker inc. not wanting to introduce anything like this into production containers all over the world.

They didn't have any problem quietly introducing a hidden DNS server so I wouldn't put my money on it. The argument of the different platform versions being for different usage is a red herring. The whole point of the Mac and Win versoins is mimicking the Linux version to avoid ugly surprises when deploying to production.

Reply:
Are we in a situation where current Linux users do not really use XDebug?

If so making a change which only targets Docker for Mac users is primarily a matter of not messing it up for Linux users. Our current usage relies on `docker-compose-dev.yml` where `xdebug.remote_host` and `xdebug.remote_connect_back` is set appropriately. This is then loaded automatically when using `docker-sync`. Avoiding that implicit dependency would be nice.

On the other hand I would appreciate a situation where XDebug was available on both Linux and Docker for Mac. Perhaps that just requires different setups for each platform?

Reply:
Ok, did a force push. I now set remote_host if the user is not already providing one, and I can look up the hostname.

Reply:
@kasperg 
> Are we in a situation where current Linux users do not really use XDebug?

I would say no. XDebug usage isn't as much a platform issue, as an issue of IDE. Those using PHPStorm or whatever that has an integrated debugger that works with XDebug is likely using them, so they'd probably like to not be left out in the cold.

Reply:
@xendk My comment was referring to current Linux users *at Reload* 😄 . I am sure vim and Emacs has debugger plugins that works with XDebug.

Reply:
@kasperg 
So was I. I think @achton uses XDebug debugging, and I know @naxoc would rather like to. If looking for correlation, being an old fart with a daughter seems a better fit than IDE usage, as @arnested and I seems to be the ones that mostly can't be bothered with it.